src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machines = new ArrayList<>(new BulldozerProducer().get());
+        } else if (type.equals(Excavator.class)) {
+            machines = new ArrayList<>(new ExcavatorProducer().get());
+        } else if (type.equals(Truck.class)) {
+            machines = new ArrayList<>(new TruckProducer().get());
+        }
```

Comment:
add field in service Map with Machine class(Bulldozer.class, Truck.class) as a key and MachineProducer as value
initialize it in initialization block or constructor

Reply:
I don't know anything about the map yet;)